### DISCUSSION TITLE
Changing parameter bounds for an existing study

### DISCUSSION THREAD

**Discussion by JCCPort** [optuna/optuna#5505] [Q&A]
Hi, I've got a study that I've left to optimise for a very long time so have a lot of trials completed already. I want to loosen the bounds of some parameters but have the sampler still able to draw from information in the existing study - or at least have the trials tidily in the same place. I've played around with this a bit and it seems sometimes I can change a parameter bound and re-run and the sampler will start to explore the new region of parameter space, but other times this doesn't happen. Is there a way to guarantee that the sampler uses the new bounds?

**Comment by HideakiImamura** [comment]
Optuna is based on the define-by-run concept, which means the search space is dynamically constructed in the objective function. Therefore, you can continue the optimization if you fix your objective function, in which the bounds of some parameters are loosen. Note that if you make the bounds tighter then the sampling algorithm of Optuna faces the illegal value of some parameters, e.g. the new bounds is [0, 1] but some trials' parameter values are outside of it. So if you want to change the bounds of parameters, you should always make it "looser".
